src/pages/organization/Layers.vue

```diff
@@ -47,93 +47,106 @@
   </div>
 </template>
 
-<script>
-import LayerUploadTool from '@/components/LayerUploadTool';
-import LocationTable from '../../components/LocationTable';
+<script lang="ts">
+import LayerUploadTool from '@/components/LayerUploadTool.vue';
+import LocationTable from '@/components/LocationTable.vue';
 import User from '../../models/User';
 import Location from '../../models/Location';
 import LocationType from '../../models/LocationType';
 import { getQueryString } from '../../utils/urls';
 import { getErrorMessage } from '../../utils/errors';
+import { useToast } from 'vue-toastification';
 
 export default {
   name: 'Layers',
   components: { LocationTable, LayerUploadTool },
-  data() {
-    return {
-      locations: [],
-      locationTypeFilter: null,
-      currentSearch: '',
-      locationsLoading: false,
-      locationsMeta: {
+  setup() {
+    const store = useStore();
+    const toasted = useToast();
+    const { t } = useI18n();
+
+    const currentUser = computed(() => User.find(store.getters['auth/userId']));
+    const locationTypes = computed(() => LocationType.all());
+    const locationTypeFilter = ref();
+    const currentSearch = ref<string>('');
+    const locationsLoading = ref<Boolean>(false);
+    const locations = ref<Array<Location>>([]);
+    const locationsMeta = ref({
```

Comment:
It makes more sense to use `reactive` here instead of ref because the value is an object
```suggestion
    const locationsMeta = reactive({
```